RFC/src/RFC-0345_AssetLifeCycle.md

```diff
@@ -0,0 +1,145 @@
+  * The original asset SHOULD be marked as `retired`. If so, the `superseded_by` field in the old asset will carry the
+    new asset id once the new asset has been confirmed. We recommend retiring the old asset because all the keys that
+    indicate ownership of tokens will be copied over; effectively re-using them; which can damage privacy.
+  * A policy is provided to determine the course of action to follow if any state from the old asset is illegal under
+    the new template rules (e.g. If a new rule requires `token.age` to be > 21; what happens to any tokens where this
+    condition fails?)
+
+
+As part of the migration,
+
+1. An entirely new asset is created with the full state of the old asset copied over into the new asset; supplemented
+   with any additional state required in the new template.
+2. A state validation run is performed; and any invalid state from the old asset is modified according to the migration
+  policy.
+3. Step 2 is repeated until a valid initial state is produced.
+4. If Step 2 has run `STATE_VALIDATION_LIMIT` times and the initial state is still not valid, the migration instruction
+   is rejected; the migrate_asset instruction will advise what should be done with the original asset in this case:
+   either allow the original asset to continue as before, or retire it.
+5. Once a valid initial state is produced, a new `create_asset` instruction is generated from the initial
+   state and the `migrate_asset` instruction. Typically the same VN committee will be used for the new asset, but this
+   needn’t be the case.
+
+Once a successful migration has completed, any instructions to the old asset can return a simple `asset_migrated`
+response with the new asset ID, which will allow clients and wallets to seamlessly update their records.
+
+### Retiring Assets
+
+Retiring an asset follows the same procedure as when as asset reaches its natural end-of-life: A final checkpoint is
+posted to the base layer and a grace period is given to allow DAN nodes and clients to take a snapshot of the final
+state if desired.
+
+### Resurrecting assets
+
+It’s unreasonable to expect VNs to hold onto large chunks of state for assets that are effectively dead (e.g. ticket
+stubs long after the event is over). For this reason, assets are allowed to expire after which VNs can forget about the
+state and use that storage for something else.
+
+However, it may be that interest in an asset resurfaces long after the asset expires (nostalgia being the multi-billion
+dollar industry it is today). The `resurrect_asset` instruction provides a mechanism to bring an asset back to life.
+
+To resurrect an asset, the following conditions must be met:
+
+* The asset must have expired.
+* It must not be currently active (i.e. it hasn’t already been resurrected).
+* An asset issuer (not necessarily the original asset issuer) must provide funding for the new lifetime of the asset.
```

Comment:
Are there any rules or restrictions governing a new (different) asset issuer resurrecting an asset? Can anyone who has the asset state and funding perform this action or must permission be given by the owner?

Reply:
I'd think anyone with the requisite data can; but open to hearing opinions on this point.

Reply:
I kinda see it as letting your domain name registration lapse. It goes back into the public domain after that

Reply:
I'm actually wondering how you might prove that you have the final state for resurrecting a contract. You'd actually need to trust the new VN committee, right? They could *not* have the state and just present the Merkle root off the blockchain as "evidence"; leading to a whole host of attacks. Let's say Alice wants to steal from Bob. She resurrects a contract using a modified VN instance that will allow her to "sell" an asset to Bob and take his money without actually having the ability to effect the transfer.

So maybe the original asset issuer should have to give permission.

Reply:
But how about we ask the new "VN committee" to prove it knows the state? We can ask them to provide us with the merkle proof of a bunch of random assets in the state of "our" choosing. 

That way the committee doesnt have to provide the complete state as that might be very large. But a bunch of random states should suffice for security reasons that they know the state.

Reply:
Great suggestion.